### PR TITLE
enh : accept float data as int field input

### PR DESCRIPTION
**Pull request by bobliubj**
For integer fields, users are allowed to pass in floating point numbers, hexadecimal numbers, and binary numbers.
For floating-point numbers, rounding is used as the field result value.
If it is converted to an integer and exceeds the field expression range, an error is reported

### CODE REVIEW

File: source/common/src/tvariant.c [hunk 1]
```diff
@@ -19,17 +19,215 @@
...
+
+    if (z[1] == 'x' || z[1] == 'X') {
+      // parsing as hex
+      return parseHexUInteger(z, n, value);
+    }
+  }
+
+  // rm flag u-unsigned, l-long, f-float(if not in hex str)
+  char last = tolower(z[n - 1]);
+  if (last == 'u' || last == 'l' || last == 'f') {
```
Comment (dapan1121) [inline source/common/src/tvariant.c]: 支持这几个字符的意义是什么？

File: source/common/src/tvariant.c [hunk 2]
```diff
@@ -19,17 +19,215 @@
...
+
+int32_t parseSignAndUInteger(const char *z, int32_t n, bool *is_neg, uint64_t *value) {
+  *is_neg = false;
+  if (n < 1) {
+    return TSDB_CODE_FAILED;
+  }
+
+  // parse sign
+  bool has_sign = false;
+  if (z[0] == '-') {
```
Comment (dapan1121) [inline source/common/src/tvariant.c:51]: 浮点数写入支持吗

File: source/common/src/tvariant.c [hunk 3]
```diff
@@ -19,17 +19,215 @@
...
+    return TSDB_CODE_FAILED;
+  }
+  *value = round(val);
+  return TSDB_CODE_SUCCESS;
+}
+
+int32_t removeSpace(const char **pp, int32_t n) {
+  // rm blank space from both head and tail
+  const char *z = *pp;
+  while (*z == ' ' && n > 0) {
```
Comment (dapan1121) [inline source/common/src/tvariant.c]: 其他字符比如换行、TAB呢

File: tests/system-test/1-insert/insert_double.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        self.db = "db1"
+        tdLog.debug(f"start to excute {__file__}")
+        tdSql.init(conn.cursor(), False)
+
+    def prepare_db(self):
+        tdSql.execute(f"drop database if exists {self.db}")
+        tdSql.execute(f"create database {self.db}")
+        tdSql.execute(f"use {self.db}")
+
+    def test_value(self, table_name, dtype, bits):
```
Comment (dapan1121) [inline tests/system-test/1-insert/insert_double.py:25]: 缺少异常用例

**Review by dapan1121 — COMMENTED** [review]
1. 检查哪些规则有必要，是否需要应用到其他类型的写入，比如+、-、空格那浮点数写入是不是也要支持？
2. 测试一下正常写入的性能影响，比如10列整数写入场景

File: source/common/src/tvariant.c (hunk 2 above)
Comment (bobliubj) [reply source/common/src/tvariant.c:51]: 支持

File: source/common/src/tvariant.c (hunk 3 above)
Comment (bobliubj) [reply source/common/src/tvariant.c]: 尾部space若不清理，写入时提示字符串非法。其他字符过滤暂不支持

**Pull request merged**